graftm/hmmer.py

```diff
@@ -598,63 +592,87 @@ def p_search(self, files, args, run_stats, base, unpack, raw_reads):
                       restrict_read_length=args.restrict_read_length)
         extracting_orfm = OrfM(min_orf_length=args.min_orf_length,
                       restrict_read_length=args.restrict_read_length)
-
-        hit_table = self.hmmsearch(files.hmmsearch_output_path(base),
-                                   raw_reads,
-                                   unpack,
-                                   args.input_sequence_type,
-                                   args.threads,
-                                   args.eval,
-                                   orfm)
-        # Processing the output table to give you the readnames of the hits
-        run_stats, hit_readnames = self.csv_to_titles(files.readnames_output_path(base),
-                                                      hit_table,
-                                                      run_stats)
+        
+        if search_method == 'hmmsearch':
+            search_result = self.hmmsearch(
+                                           hmmsearch_output_table,
+                                           raw_reads,
+                                           unpack,
+                                           args.input_sequence_type,
+                                           args.threads,
+                                           args.eval,
+                                           orfm
+                                           )
+            for result in search_result:
+                hit_readnames=[x[0] for x in result.each([SequenceSearchResult.QUERY_ID_FIELD])]
+        
+        elif search_method == 'diamond':
+            #run diamond
+            search_result =  Diamond(
+                                     database=os.path.join(
+                                                           gpkg._base_directory, \
+                                                           gpkg._contents_hash[gpkg.DIAMOND_DATABASE_KEY]
```

Comment:
Don't use single underscore methods from outside the class they are defined in. Here there is a method for getting the diamond database